browser/src/communication/federated/federated_client.ts

```diff
@@ -119,13 +123,34 @@ export class FederatedClient extends Client {
     }
   }
 
+  private addDifferentialPrivacy (currentRoundWeights: Weights): Weights {
```

Comment:
make state-less, by also giving `previousRoundWeights = this.previousRoundWeights` as a param.

Reply:
I agree on the benefits of stateless-ness, however `client` is the class actually updating the `trainer`'s model, hence the one which has direct access to both the previous and current model. Moving the state to `trainer` would require more effort and might not be cleaner 🤔